filer/models/filemodels.py

```diff
@@ -241,6 +241,19 @@ def url(self):
         return r
 
     @property
+    def canonical_url(self):
+        url = ''
+        if self.file and self.is_public:
+            try:
+                url = urlresolvers.reverse('canonical', kwargs={
+                    'uploaded_at': self.uploaded_at.strftime('%s'),
```

Comment:
We can use the `File.uploaded_at` field, which is non-editable, to make the public file's URL a little bit more hard to guess.